### ISSUE TITLE
pycommands: error importing module (

### ISSUE THREAD

**Issue by tee-lake** [corelan/mona#28]
### Expected behavior 
Mona should spit back data when launched (!mona) from the Immunity Debugger 1.85

### Actual behavior
pycommands: error importing module
....
line 87 in <module>
import urllib ...
Error Image here
![image](https://user-images.githubusercontent.com/25264165/44483823-28b63800-a65d-11e8-8ee1-053d5ae45347.png)

### Steps to reproduce the problem
installed Immunity Debugger and updated from python 2.7.1 to python 2.7.14
added mona.py to the PyCommands folder
started Immunity Debugger as an admin
use !mona
### Other useful information (mona version, debugger & debugger version, OS version, etc)
OS: Windows 7 x64 SP 1
immunity debugger 1.85

**Comment by tee-lake** [comment]
I downgraded the python version back to 2.7.1 which was installed by Immunity Debugger and it worked!

**Comment by corelanc0d3r** [comment]
would it be possible to double check that you are installing the x86 version (and not the x64 version) of python 2.7.14 (or higher), and that you are installing it in the default folder (where 2.7.1 resides)?

thanks !

**Comment by sibios** [comment]
I was having this issue as well.  I can confirm that this is a result of using the x64 version of Python (2.7.15 in my case).  Installing the x86 version resolved the error.

**Comment by corelanc0d3r** [comment]
thanks - I'm also quite sure that this is caused by installing the x64 version
I'll close the issue for now, feel free to reopen if the issue has not been resolved.
(you should not be running the python 2.7.1 version, as you won't be able to run !mona upgrade)

**Comment by dcliss** [comment]
guys i get the same error

**Comment by dcliss** [comment]
which version to run ?

**Comment by corelanc0d3r** [comment]
you need to run the x86 python version, not the x64 version

**Comment by dcliss** [comment]
i installed x86 version https://www.python.org/ftp/python/2.7.14/python-2.7.14.msi

still getting the same error any help ?

**Comment by KweonKang** [comment]
You'd rather check mona.py file. It is 630,068 byte size, not 5MB.
You can get it from https://raw.githubusercontent.com/corelan/mona/master/mona.py

**Comment by freis** [comment]
Added the 32bit python.exe (version 2.7.16)binary to PATH and used this last mona.py, still getting the same error.
OS is Windows 2008 Server.

**Comment by corelanc0d3r** [comment]
@freis are you using Immunity Debugger?  Did you install the python version that comes with immunity, and installed it in the default path?   After that, you upgraded it to 2.7.16?  What was the python installer filename?

When you open mona.py, does it contain python code? How many lines does it have?

Can you run any of the other pycommands in Immunity?

**Comment by pope-es** [comment]
I am experiencing the same error on two 32-bit VMs (XP and win7). On both, I made sure I'm running 32-bit python (v 2.7.17).
In the Win7, I even:
- removing this Python version and instead installing the 2.7.1 that Immunity installs on setup. No luck either.
- ensured c:\Python27 is added to $PATH.

**Comment by corelanc0d3r** [comment]
can you please check/confirm that mona.py contains python code?
how many lines does it have ?

**Comment by pope-es** [comment]
Yep, it's 19102 lines of pure Python code. It's the same copy of the file that I am using in a Win 10 VM. All the VMs are freshly installed 32-bits versions of windows, with 2 cores and 2 GB RAM, running on VMware Fusion.

- Tried downgrading to a random earlier version of mona.py.

- Tried with Python 2.7.1 and 2.7.16, apart form the latest 2.7.17. All of them 32-bits.

Is there any way I can provide further debug info?

**Comment by corelanc0d3r** [comment]
can you check the properties of the file and confirm the file has been "unblocked"?  Files downloaded from the internet are marked as unsafe by default

can you run other pycommands (check the pycommands folders) - that should confirm that the python setup is correct

**Comment by pope-es** [comment]
can you check the properties of the file and confirm the file has been "unblocked"? Files downloaded from the internet are marked as unsafe by default -> In the file properties I don't see anything suggesting it's blocked or not trusted. In any case I also tried re-creating the file contents to a new file created by the admin, same result.

can you run other pycommands (check the pycommands folders) - that should confirm that the python setup is correct --> This. You are right, the others (pyshell, cmdcli) throw the same error, hence this is not a Mona issue. I will keep looking for my solution!

**Comment by PSYs0s** [comment]
I tried so many times in vmware. But doesn't work

**Comment by corelanc0d3r** [comment]
> I tried so many times in vmware. But doesn't work

would it be possible to provide more detailed information?
What exactly have you tried and what doesn't work?

**Comment by rodri3d** [comment]
If you look at the description in the download link, it says: "removed some unneeded logging". Once you have the file, you must 'clean it' of unnecessary characters (otherwise it'll not work at all)

cheers!

**Comment by corelanc0d3r** [comment]
if you're not able to get mona.py to run, please open a new issue

**Comment by topaz0hisham** [comment]
![image](https://user-images.githubusercontent.com/87389437/126677982-6101292b-16b1-4a2c-9e82-8114d6cd1a05.png)

**Comment by Miracl3xt** [comment]
PyCommands: error importing module

Steps to Resolve the problem
1)Download mona.py from 
https://github.com/corelan/mona

2) add mona.py to the PyCommands folder (  C:\Program Files (x86)\Immunity Inc\Immunity Debugger\PyCommands  )
start Immunity Debugger
use !mona

**Comment by iprincs** [comment]
used capital M in command
